1_IGT_PP/Code/Stan/igt_selfreport_orl_binomial_joint_indMu.stan

```diff
@@ -0,0 +1,339 @@
+functions {
+  // to compute empirical correlation matrix given input matrix
+  matrix empirical_correlation(matrix X) {
+    int N = rows(X);
+    int M = cols(X);
+    matrix[N, M] centered;
+    vector[M] std_devs;
+    matrix[M, M] Sigma;
+    matrix[M, M] R;
+
+    // calculate covariance matrix
+    for (j in 1:M) {
+      centered[,j] = X[,j] - mean(X[,j]);
+    }
+    Sigma = (centered' * centered) / (N - 1);
+
+    // Calculate correlation matrix
+    std_devs = sqrt(diagonal(Sigma));
+    for (i in 1:M) {
+      for (j in 1:M) {
+        R[i,j] = Sigma[i,j] / (std_devs[i] * std_devs[j]);
+      }
+    }
+
+    return R;
+  }
+
+  real Phi_approx_group_mean_rng(real mu, real sigma, int n_samples) {
+    vector[n_samples] par;
+    for (i in 1:n_samples) {
+      par[i] = Phi_approx(normal_rng(mu, sigma));
+    }
+    return mean(par);
+  }
+
+  matrix Phi_approx_corr_rng(vector mu, vector sigma, matrix R, int n_samples) {
+    int N = rows(R);
+    int M = cols(R);
+    matrix[N,M] Sigma = quad_form_diag(R, sigma);
+    matrix[n_samples, M] X_sim;
+    for (i in 1:n_samples) {
+      X_sim[i,] = Phi_approx(multi_normal_rng(mu, Sigma)');
+    }
```

Comment:
I realized that this will actually need changed now—the `Phi_approx` part is only necessary for parameters that use that transformation in the `transformed parameters` block. This solution is a bit ugly, but will transform the right parameters (`Arew`, `Apun`, `K`, and `theta`) to then estimate the correlation on the transformed scale the right way: 
```suggestion
    for (i in 1:n_samples) {
      X_sim[i,] = multi_normal_rng(mu, Sigma)';
    }
    // apply transformation as needed
    X_sim[,1] = Phi_approx(X_sim[,1]);
    X_sim[,2] = Phi_approx(X_sim[,2]);
    X_sim[,3] = Phi_approx(X_sim[,3]);
    X_sim[,6] = Phi_approx(X_sim[,6]);
```